### ISSUE TITLE
Custom repository class on GridFS File XML mapping is not used

### ISSUE THREAD

**Issue by webdevilopers** [doctrine/mongodb-odm#2141]
|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.0.3

While migrating to v*2* we are changing the mapping for files to the new GridFS usage.

```xml
<doctrine-mongo-mapping xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                        xmlns="http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping"
                        xsi:schemaLocation="http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping
                        http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping.xsd">

    <gridfs-file name="Acme\Foo" bucket-name="foos"
                 repository-class="CustomFooRepository">
        <id field-name="id" type="bin_uuid" />
        <length />
        <chunk-size />
        <upload-date />
        <filename field-name="filename" />
    </gridfs-file>
</doctrine-mongo-mapping>
```

But when checking the class for the used repository the custom repository is not used:

```php
$repo = $dm->getRepository(Foo::class);
dump($repo); // customRepositoryClassName: null

$meta = $dm->getMetadataFactory()->getMetadataFor(Foo:class);
dump($meta); // customRepositoryClassName: null
```

Possibly related:
* https://github.com/doctrine/mongodb-odm/pull/1790

Came from:
* https://twitter.com/webdevilopers/status/1209449773561982977

**Comment by webdevilopers** [comment]
This seems to be an issue with the XML mapping.
I tried setting the metadata manually according to the tests and it worked.

```php
$meta = $dm->getMetadataFactory()->getMetadataFor(Foo:class);
$meta->setCustomRepositoryClass(CustomFooRepository:.class);
dump($meta); // customRepositoryClassName: null
```

* https://github.com/doctrine/mongodb-odm/blob/master/tests/Doctrine/ODM/MongoDB/Tests/Mapping/ClassMetadataTest.php#L51

**Comment by alcaeus** [comment]
Thanks for the report. This will be fixed in 2.0.4.

**Comment by alcaeus** [comment]
Fixed in #2146.
